mr/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergTestUtils.java

```diff
@@ -0,0 +1,180 @@
+
+  public static final StandardStructObjectInspector FULL_SCHEMA_OBJECT_INSPECTOR =
+      ObjectInspectorFactory.getStandardStructObjectInspector(
+          // Capitalized `boolean_type` field to check for field case insensitivity.
+          Arrays.asList("Boolean_Type", "integer_type", "long_type", "float_type", "double_type",
+              "date_type", "tsTz", "ts", "string_type", "uuid_type", "fixed_type", "binary_type", "decimal_type"),
+          Arrays.asList(
+              PrimitiveObjectInspectorFactory.writableBooleanObjectInspector,
+              PrimitiveObjectInspectorFactory.writableIntObjectInspector,
+              PrimitiveObjectInspectorFactory.writableLongObjectInspector,
+              PrimitiveObjectInspectorFactory.writableFloatObjectInspector,
+              PrimitiveObjectInspectorFactory.writableDoubleObjectInspector,
+              IcebergObjectInspector.DATE_INSPECTOR,
+              IcebergObjectInspector.TIMESTAMP_INSPECTOR_WITH_TZ,
+              IcebergObjectInspector.TIMESTAMP_INSPECTOR,
+              PrimitiveObjectInspectorFactory.writableStringObjectInspector,
+              PrimitiveObjectInspectorFactory.writableStringObjectInspector,
+              IcebergBinaryObjectInspector.byteArray(),
+              IcebergBinaryObjectInspector.byteBuffer(),
+              IcebergDecimalObjectInspector.get(38, 10)
+          ));
+
+  private HiveIcebergTestUtils() {
+    // Empty constructor for the utility class
+  }
+
+  public static Record getTestRecord(boolean uuidAsByte) {
+    Record record = GenericRecord.create(HiveIcebergTestUtils.FULL_SCHEMA);
+    record.set(0, true);
+    record.set(1, 1);
+    record.set(2, 2L);
+    record.set(3, 3.1f);
+    record.set(4, 4.2d);
+    record.set(5, LocalDate.of(2020, 1, 21));
+    // TimeType is not supported
+    // record.set(6, LocalTime.of(11, 33));
+    // Nano is not supported ?
+    record.set(6, OffsetDateTime.of(2017, 11, 22, 11, 30, 7, 0, ZoneOffset.ofHours(2)));
+    record.set(7, LocalDateTime.of(2019, 2, 22, 9, 44, 54));
+    record.set(8, "kilenc");
+    if (uuidAsByte) {
+      // TODO: Parquet UUID expect byte[], others are expecting UUID
+      record.set(9, UUIDUtil.convert(UUID.fromString("1-2-3-4-5")));
+    } else {
+      record.set(9, UUID.fromString("1-2-3-4-5"));
+    }
+    record.set(10, new byte[]{0, 1, 2});
+    record.set(11, ByteBuffer.wrap(new byte[]{0, 1, 2, 3}));
+    record.set(12, new BigDecimal("0.0000000013"));
+
+    return record;
+  }
+
+  public static Record getNullTestRecord() {
+    Record record = GenericRecord.create(HiveIcebergTestUtils.FULL_SCHEMA);
+
+    for (int i = 0; i < HiveIcebergTestUtils.FULL_SCHEMA.columns().size(); i++) {
+      record.set(i, null);
+    }
+
+    return record;
+  }
+
+  public static List<Object> valuesForTestRecord(Record record) {
+    ByteBuffer byteBuffer = record.get(11, ByteBuffer.class);
+    byte[] bytes = new byte[byteBuffer.remaining()];
```

Comment:
You should be able to use utils in `ByteBuffers` for this.

Reply:
Done.
Thanks!